rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,284 @@
+        token_user_amount = user_amount / total_amount * token_total_amount
+        weight = token_weight * 100 / total_weight
+        pool_token = BalancerPoolToken(
+            token=token,
+            total_amount=token_total_amount,
+            user_balance=Balance(amount=token_user_amount),
+            weight=weight,
+        )
+        pool_tokens.append(pool_token)
+
+    pool = BalancerPool(
+        address=address,
+        tokens=pool_tokens,
+        total_amount=total_amount,
+        user_balance=Balance(amount=user_amount),
+    )
+    return user_address, pool
+
+
+def deserialize_transaction_id(raw_tx_id: str) -> Tuple[str, int]:
+    try:
+        tx_hash, raw_log_index = raw_tx_id.split('-')
+        log_index = int(raw_log_index)
+    except ValueError as e:
+        raise DeserializationError(f'Unexpected transaction id: {raw_tx_id}') from e
+    return tx_hash, log_index
+
+
+def deserialize_swap(raw_swap: Dict[str, Any]) -> AMMSwap:
+    """May raise DeserializationError"""
+    attr_raw_addresses: Dict[str, str] = {}  # not checksum addresses
+    attr_addresses: Dict[str, ChecksumEthAddress] = {}  # checksum addresses
+    try:
+        tx_hash, log_index = deserialize_transaction_id(raw_swap['id'])
+        timestamp = deserialize_timestamp(raw_swap['timestamp'])
+        raw_tokens = raw_swap['poolAddress']['tokens']
+        token0_symbol = raw_swap['tokenInSym']
+        token1_symbol = raw_swap['tokenOutSym']
+        amount0_in = deserialize_asset_amount(raw_swap['tokenAmountIn'])
+        amount1_out = deserialize_asset_amount(raw_swap['tokenAmountOut'])
+        attr_raw_addresses['userAddress'] = raw_swap['userAddress']['id']  # address
+        attr_raw_addresses['caller'] = raw_swap['caller']  # from_address
+        attr_raw_addresses['poolAddress'] = raw_swap['poolAddress']['id']  # to_address
+        attr_raw_addresses['tokenIn'] = raw_swap['tokenIn']  # token0_address
+        attr_raw_addresses['tokenOut'] = raw_swap['tokenOut']  # token1_address
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    # Checksum addresses
+    for attr, raw_address in attr_raw_addresses.items():
+        try:
+            attr_addresses[attr] = to_checksum_address(raw_address)
+        except ValueError as e:
+            raise DeserializationError(
+                f'Invalid ethereum address: {raw_address} in swap {attr}.',  # noqa: E501
+            ) from e
+
+    # Get token0 and token1
+    # Retrieve name and decimals just in case we have UnknownEthereumToken
+    # TODO VN: it seems there may exist pools without tokens... the swap contains
+    # the IN/OUT tokens but we can't get the name and decimalswhich means
+    # we can get all the name/decimals for the UnknownEthereumToken
+    try:
+        raw_address_tokens = {raw_token['address']: raw_token for raw_token in raw_tokens}
+        raw_token0 = raw_address_tokens[attr_raw_addresses['tokenIn']]
```

Comment:
I've found this problem implementing the integration test for `get_trades_history` with the address of [this transaction](0x70b50d011f23f8437aa32b4155972098b34b5fb19b02198f3bc7f2456c4c3ffc). I will address it tomorrow.